File: Dormitories/ClientApp/app/room/Room.ts
import { Faculty } from '../faculty/Faculty';
import { Floor } from '../floor/Floor';
import { Block } from '../block/Block';
import { Student } from '../student/Student';

export class Room {
    id: number;
    name: string;
    totalPlaces: number;
    freePlaces: number;
    faculty: Faculty;
    floor: Floor;
    block: Block;
    students: Student[];
}

File: Dormitories/Models/Room.cs
using System.Collections.Generic;

namespace Dormitories.Models
{
    public class Room
    {
        public int Id { get; set; }
        public int TotalPlaces { get; set; }
        public int FreePlaces { get; set; }
        public Faculty Faculty { get; set; }
        public Floor Floor { get; set; }
        public Block Block { get; set; }
        public string Name { get; set; }
        public List<Student> Students { get; set; }
    }
}